plugins/main/public/components/overview/azure/dashboards/dashboard.tsx

```diff
@@ -0,0 +1,153 @@
+import { compose } from 'redux';
+import { getPlugins } from '../../../../kibana-services';
+import { ViewMode } from '../../../../../../../src/plugins/embeddable/public';
+import { getDashboardPanels } from './dashboard_panels';
+import { I18nProvider } from '@osd/i18n/react';
+import { SearchResponse } from '../../../../../../../src/core/server';
+import './styles.scss';
+import { withErrorBoundary } from '../../../common/hocs';
+import { IndexPattern } from '../../../../../../../src/plugins/data/common';
+import { SampleDataWarning } from '../../../visualize/components';
+import {
+  ErrorFactory,
+  ErrorHandler,
+  HttpError,
+} from '../../../../react-services/error-management';
+import {
+  AlertsDataSourceRepository,
+  PatternDataSource,
+  tParsedIndexPattern,
+  useDataSource,
+  AzureDataSource,
+} from '../../../common/data-source';
+import { DiscoverNoResults } from '../../../common/no-results/no-results';
+import { LoadingSearchbarProgress } from '../../../common/loading-searchbar-progress/loading-searchbar-progress';
+import { useReportingCommunicateSearchContext } from '../../../common/hooks/use-reporting-communicate-search-context';
+import { WzSearchBar } from '../../../common/search-bar';
+import { WAZUH_SAMPLE_ALERTS_CATEGORY_SECURITY } from '../../../../../common/constants';
+
+const plugins = getPlugins();
+
+const DashboardByRenderer = plugins.dashboard.DashboardContainerByValueRenderer;
+
+const DashboardAzureComponents: React.FC = ({}) => {
+  const AlertsRepository = new AlertsDataSourceRepository();
+  const {
+    filters,
+    dataSource,
+    fetchFilters,
+    fixedFilters,
+    isLoading: isDataSourceLoading,
+    fetchData,
+    setFilters,
+  } = useDataSource<tParsedIndexPattern, PatternDataSource>({
+    DataSource: AzureDataSource,
+    repository: AlertsRepository,
+  });
+
+  const [results, setResults] = useState<SearchResponse>({} as SearchResponse);
+
+  const { searchBarProps, fingerprint, autoRefreshFingerprint } = useSearchBar({
+    indexPattern: dataSource?.indexPattern as IndexPattern,
+    filters,
+    setFilters,
+  });
+
+  const { query, dateRangeFrom, dateRangeTo } = searchBarProps;
+
+  useReportingCommunicateSearchContext({
+    isSearching: isDataSourceLoading,
+    totalResults: results?.hits?.total ?? 0,
+    indexPattern: dataSource?.indexPattern,
+    filters: fetchFilters,
+    query: query,
+    time: { from: dateRangeFrom, to: dateRangeTo },
+  });
+
+  useEffect(() => {
+    if (isDataSourceLoading) {
+      return;
+    }
+    fetchData({
+      query,
+      dateRange: { from: dateRangeFrom, to: dateRangeTo },
+    })
+      .then(results => setResults(results))
+      .catch(error => {
+        const searchError = ErrorFactory.create(HttpError, {
+          error,
+          message: 'Error fetching data',
+        });
+        ErrorHandler.handleError(searchError);
+      });
+  }, [
+    JSON.stringify(fetchFilters),
```

Comment:
This may come at a cost of few more re-renders - if the object props come in a different order, but with the same values it may fire a re-render
It's a good practice to memoize this kind of objects and pass them in the dependency array afterwards
e.g: 
`const fiilters = useMemo(() => fetchFilters, [JSON.stringify(fetchFilters)]`
useEfect dependency array: 
`[fiilters, ...otherdependencies]`

Feel free to dismiss my review: I m practicing my review skills and wazuh seems like an interesting project where I can contribute

Reply:
Good observation, this should be changed in another issue because it is done in all dashboards the same and should be changed in all, to keep a pattern for now we will leave it like this and in the future we will change all of them.